src/WFQC/data.py

```diff
@@ -149,82 +150,95 @@ async def get_biotools_metadata(outpath, topicID="topic_0121"):  # TODO: I remov
             else: 
                 print(f'Error while fetching tool names from page {page}')
                 break
-    
-    print("Nr of tools without a primary publication tag:", no_primary_publications)
-    print("Largest number of publications for a tool: ",max(nr_publications))
-    print("Nr of tools with pmid in bio.tools: ",len(all_tool_data))
-    print("Nr of tools without pmid (with doi): ", len(doi_tools))
 
-    # Download pmids from dois
-    doi_library_filename = 'doi_pmid_library.json' # TODO: Make it customisable 
-    try: 
-        with open(doi_library_filename, 'r') as f:
-            doi_library = json.load(f)
-    except FileNotFoundError:
-        print(f'Library file not found. Creating new file named {doi_library_filename}.')
-        doi_library = {} # {doi: pmid}, should I perhaps do {name: [pmid doi ]} instead?
+    # Record the total nr of tools
+    nr_tools = 0  
+    if biotool_data: 
+        nr_tools = int(biotool_data['count']) 
 
-    library_updates = False
-    async with aiohttp.ClientSession() as session: 
-        for tool in tqdm(doi_tools, desc="Fetching pmids from dois."):
-            doi = tool["doi"]
+    return pmid_tools, doi_tools, no_primary_publications, nr_publications, nr_tools
 
-            # Check if tool is already in library 
-            if doi in doi_library: 
-                doi_pmid = doi_library[doi] 
-                tool["pmid"] = doi_pmid
-                continue
-            
-            # Otherwise access NCBI API
-            library_updates = True
+def get_tool_metadata(outpath, topicID="topic_0121"):  # TODO: I removed format. Check if there is any reason to have it 
```

Comment:
I suggest extracting methods like this one to the WFQC package, usually notebooks are nice for demonstrating code, but they should not be too long either. 

Note: It is a bit difficult to make a review because of the number of files in the PR, the whole page is lagging. I will not make more inline comments and will leave you some general feedback.